File: apps/web/src/lib/colors.ts
export const Color = {
	Red: 0,
	Blue: 1,
	Black: 2,
	Green: 3,
	Yellow: 4,
	Purple: 5,
	White: 6,
	Orange: 7,
	Cyan: 8
} as const;

export type ColorId = (typeof Color)[keyof typeof Color];

export function colorToCss(color: ColorId): string {
	switch (color) {
		case Color.Red:
			return '#ef4444';
		case Color.Blue:
			return '#3b82f6';
		case Color.Black:
			return '#0f172a';
		case Color.Green:
			return '#22c55e';
		case Color.Yellow:
			return '#eab308';
		case Color.Purple:
			return '#a855f7';
		case Color.White:
			return '#f8fafc';
		case Color.Orange:
			return '#f97316';
		case Color.Cyan:
			return '#06b6d4';
		default:
			return '#94a3b8';
	}
}

File: crates/engine/src/colors.rs
use serde::{Deserialize, Serialize};

pub const COLOR_COUNT: usize = 9;

/// 颜色编码（与 `rules.json` 保持一致）。
#[repr(u8)]
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
pub enum Color {
    Red = 0,
    Blue = 1,
    Black = 2,
    Green = 3,
    Yellow = 4,
    Purple = 5,
    White = 6,
    Orange = 7,
    Cyan = 8,
}

impl Color {
    pub fn from_u8(v: u8) -> Option<Self> {
        Some(match v {
            0 => Self::Red,
            1 => Self::Blue,
            2 => Self::Black,
            3 => Self::Green,
            4 => Self::Yellow,
            5 => Self::Purple,
            6 => Self::White,
            7 => Self::Orange,
            8 => Self::Cyan,
            _ => return None,
        })
    }

    pub fn to_u8(self) -> u8 {
        self as u8
    }
}

pub const NON_WHITE_COLORS: [Color; 8] = [
    Color::Red,
    Color::Blue,
    Color::Black,
    Color::Green,
    Color::Yellow,
    Color::Purple,
    Color::Orange,
    Color::Cyan,
];
